Sources/ArcGISToolkit/FloorFilter/FloorFilterViewModel.swift

```diff
@@ -0,0 +1,126 @@
+//
+// Copyright 2021 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import UIKit
+import ArcGIS
+
+/// View Model class that contains the Data Model of the Floor Filter
+/// Also contains the business logic to filter and change the map extent based on selected site/level/facility
+internal class FloorFilterViewModel {
+    /// The MapView, Map and Floor Manager are set in the FloorFilterViewController when the map is loaded
+    public var mapView: AGSMapView?
+    public var map: AGSMap?
+    public var floorManager: AGSFloorManager?
```

Comment:
I wanted the `FloorFilterViewModel` to have the `floorManager` so that the ViewModel will be responsible for fetching the sites/facilities/level data from the `floorManager` rather than having the `FloorFilterViewController` be responsible for the logic of getting the data from the FloorManager and then passing that down to the `FloorFilterViewModel`.